kibicara/platforms/email/webapi.py

```diff
@@ -0,0 +1,177 @@
+    :return: Email row of the found email bot.
+    """
+    try:
+        return await Email.objects.get(hood=hood)
+    except NoMatch:
+        return HTTPException(status_code=status.HTTP_404_NOT_FOUND)
+    except MultipleMatches:
+        # Email rows *should* be unique - the unique constraint doesn't work yet though.
+        if email_id is None:
+            email_id = hood.id
+        return await Email.objects.get(hood=hood, id=email_id)
+
+
+# registers the routes, gets imported in /kibicara/webapi/__init__.py
+router = APIRouter()
+
+
+@router.post('/', status_code=status.HTTP_201_CREATED)
+async def email_create(hood=Depends(get_hood)):
+    """ Create an Email bot. Call this when creating a hood.
+
+    :param hood: Hood row of the hood the Email bot is supposed to belong to.
+    :return: Email row of the new email bot.
+    """
+    try:
+        email_row = await Email.objects.create(hood=hood, secret=urandom(32).hex())
+        spawner.start(email_row)
+        return email_row
+    except IntegrityError:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT)
+
+
+@router.delete('/{email_id}', status_code=status.HTTP_200_OK)
+async def email_delete(email_id, hood=Depends(get_hood)):
+    """ Delete an Email bot. Call this when deleting a hood.
+    Stops and deletes the Email bot as well as all subscribers.
+
+    :param hood: Hood the Email bot belongs to.
+    """
+    email_row = await get_email(hood, email_id=email_id)
+    spawner.stop(email_row)
+    await EmailSubscribers.objects.delete_many(hood=hood.id)
+    await email_row.delete()
+
+
+@router.post('/subscribe/', status_code=status.HTTP_200_OK)
+async def email_subscribe(subscriber: Subscriber, hood=Depends(get_hood_unauthorized)):
+    """ Send a confirmation mail to subscribe to messages via email.
+
+    :param subscriber: Subscriber object, holds the email address.
+    :param hood: Hood the Email bot belongs to.
+    :return: Returns status code 200 after sending confirmation email.
+    """
+    token = to_token(email=subscriber.email)
+    confirm_link = (
+        config['root_url'] + "api/" + str(hood.id) + "/email/subscribe/confirm/" + token
+    )
+    try:
+        send_email(
+            subscriber.email,
+            "Subscribe to Kibicara " + hood.name,
+            sender=hood.name,
+            body="To confirm your subscription, follow this link: " + confirm_link,
+        )
+    except ConnectionRefusedError:
+        logger.info(token)
+        logger.error("Sending subscription confirmation email failed.", exc_info=True)
+        raise HTTPException(status_code=status.HTTP_502_BAD_GATEWAY)
+    except SMTPException:
+        logger.info(token)
+        logger.error("Sending subscription confirmation email failed.", exc_info=True)
+        raise HTTPException(status_code=status.HTTP_502_BAD_GATEWAY)
+
+
+@router.get('/subscribe/confirm/{token}', status_code=status.HTTP_201_CREATED)
```

Comment:
should be POST

Reply:
Not really, this link is supposed to be opened by users, in a browser, after they clicked in the mail - afaik that will result in a GET request by default, so it would return a 405 if we make it a POST request.

Reply:
Users will eventually get a link that points to a frontend site. The API should use methods with the right semantic.